plugins/clipboard-manager/src/desktop.rs

```diff
@@ -44,4 +48,35 @@ impl<R: Runtime> Clipboard<R> {
             Err(e) => Err(crate::Error::Clipboard(e.to_string())),
         }
     }
+
+    pub fn write_html(&self, kind: ClipKind) -> crate::Result<()> {
+        match kind {
+            ClipKind::Html { html, alt_html, .. } => match &self.clipboard {
+                Ok(clipboard) => clipboard.lock().unwrap().set_html(html, alt_html).map_err(Into::into),
+                Err(e) => Err(crate::Error::Clipboard(e.to_string())),
+            }
+            _ => Err(crate::Error::Clipboard("Invalid clip kind!".to_string())),
+            }
+        }
+
+    pub fn read_html(&self) -> crate::Result<ClipboardContents> {
+        match &self.clipboard {
+            Ok(clipboard) => {
+                let html = clipboard.lock().unwrap().get_text()?;
+                // arboard does not have read_html() as its just text, therefor unable to populate alt_html in this case
+                Ok(ClipboardContents::Html { html, alt_html: None })
+            }
+            Err(e) => Err(crate::Error::Clipboard(e.to_string())),
+        }
+    }
+
+    pub fn clear(&self) -> crate::Result<()> {
+        match &self.clipboard {
+            Ok(clipboard) => {
+                clipboard.lock().unwrap().clear().unwrap();
+                Ok(())
```

Comment:
We can't unwrap here, need to propagate the error
```suggestion
                clipboard.lock().unwrap().clear().map_err(Into::into)
```